src/hooks/usePendingOrders.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react'
+import { PendingTxArray } from 'api/exchange/ExchangeApi'
+
+import useGlobalState from './useGlobalState'
+import useSafeState from './useSafeState'
+import { useWalletConnection } from './useWalletConnection'
+
+function usePendingOrders(): PendingTxArray {
+  const { userAddress, networkId } = useWalletConnection()
+
+  const [{ pendingOrders: pendingOrdersGlobal }] = useGlobalState()
+  const [pendingOrders, setPendingOrders] = useSafeState<PendingTxArray>([])
+
+  useEffect(() => {
+    userAddress && networkId && setPendingOrders(pendingOrdersGlobal[networkId][userAddress] || [])
+    // updating list of orders on every block by listening on `blockNumber`
```

Comment:
You are not listening to `blockNumber`.
I know this is from `useOrders` hook, but should you be updating pending orders on every block?

Reply:
pending orders are only detected via the Trade page on submission - so checking via block update wouldn't do anything as they are auto filtered out on mining via the async place order function - unless I'm missing sth

Reply:
Then it's fine, just remove the comment :)

Reply:
:+1: